pkg/splunk/enterprise/configuration.go

```diff
@@ -756,6 +757,51 @@ func AreRemoteVolumeKeysChanged(client splcommon.ControllerClient, cr splcommon.
 	return false
 }
 
+// ValidateAppFrameworkSpec checks and validates the Apps Frame Work config
+func ValidateAppFrameworkSpec(appFramework *enterprisev1.AppFrameworkSpec) error {
+
+	//log all the inputs for AppFrameworkSpec.
+	//TBD: Future - use logging toggle switch to reduce any excessive logging.
+
+	log.Info("App Framework Inputs",
```

Comment:
Nit: Just thought about it. Should we move this log below the if check:

```
if !appFramework.FeatureEnabled {
    return nil
}
```
Reason being we can avoid the log if app framework is not enabled?

Also the format used is to create a scopedLog locally & use that.

```
scopedLog := log.WithName("ValidateAppFrameworkSpec").WithValues(
		"FeatureEnabled", appFramework.FeatureEnabled,
		"Type", appFramework.Type,
		"S3Endpoint", appFramework.S3Endpoint,
		"S3Bucket", appFramework.S3Bucket,
		"S3SecretRef", appFramework.S3SecretRef,
		"S3PollInterval", appFramework.S3PollInterval)
scopedLog.Info("Validating app framework spec")
```

Reply:
Reviewing it.

Reply:
Ok this is taken care of.